### DISCUSSION TITLE
Convert String to MDX / React component

### DISCUSSION THREAD

**Discussion by MarquesCoding** [vercel/next.js#56087] [Help]
### Summary

Hi,

So I've got some markdown (string) stored within my DB that is returned on an endpoint. What would be the best approach (because I can't figure it out) on displaying this? Is there a component that can convert the string to a markdown-ish react component? I'm currently running React & vite as my stack 🤔.

Thanks 🙏

**Comment by icyJoseph** [comment]
Do you mean something like this? https://github.com/remarkjs/react-markdown

**Reply by MarquesCoding** [reply]
Sorry, I should update the question -- I have MDX not markdown... So, this works for markdown, but not MDX

**Reply by icyJoseph** [reply]
Well, I think this one should work:

- https://mdxjs.com/packages/react/

You just gotta follow the guides.

I found a Vite plugin based of @mdx-js/react:

- https://github.com/brillout/vite-plugin-mdx

Where they refer to this example:

- https://github.com/brillout/vite-plugin-mdx/tree/master/examples/react

Which I have setup running on this Stackblitz for you to see: https://github.com/brillout/vite-plugin-mdx/tree/master/examples/react
